kuma/javascript/src/task-completion-survey.jsx

```diff
@@ -0,0 +1,177 @@
+// @flow
+import * as React from 'react';
+import { useContext, useEffect, useState } from 'react';
+import css from '@emotion/css';
+
+import CloseIcon from './icons/close.svg';
+import DocumentProvider from './document-provider.jsx';
+import GAProvider from './ga-provider.jsx';
+import LocaleProvider from './locale-provider.jsx';
+import UserProvider from './user-provider.jsx';
+import gettext from './gettext.js';
+
+const styles = {
+    notification: css({
+        position: 'fixed',
+        display: 'flex',
+        flexDirection: 'row',
+        top: 15,
+        right: 15,
+        width: 300,
+        padding: 15,
+        paddingRight: 8,
+        zIndex: 999999,
+        fontSize: 16,
+        backgroundColor: '#e4f0f5',
+        color: '#333',
+        border: 'solid #3d7e9a 2px',
+        borderRadius: 2,
+        boxShadow: '3px 3px 5px rgba(0, 0, 0, 0.25)'
+    }),
+    dismissButton: css({
+        alignSelf: 'start',
+        padding: 0,
+        borderWidth: 0,
+        backgroundColor: 'inherit'
+    }),
+    dismissIcon: css({
+        height: 18
+    })
+};
+
+// This is the name of the waffle flag that controls this notification
+const WAFFLE_FLAG = 'sg_task_completion';
+
+// This is the name of a localStorage property we use to prevent the
+// display of the notification to someone who has dismissed it or
+// responded to it within the last month
+const EMBARGO_TIMESTAMP = 'taskTracker';
+
+// This is how long the embargo lasts
+const EMBARGO_LENGTH = 1000 * 60 * 60 * 24 * 32; // 32 days
+
+// If the user dismisses or responds to the survey, then we call this
+// function to set the embargo timestamp. During this embargo period,
+// the notification will not be shown again to the user.
+function embargo() {
+    localStorage.setItem(
```

Comment:
Wondering whether it would be worth wrapping this in a `try/catch`? Not much we can do if it fails unless we want to fall back to using a cookie. I generally tend to do it, even if only to report the errors to myself in the `console` ;) There is also the issue with Safari in private mode - https://michalzalecki.com/why-using-localStorage-directly-is-a-bad-idea/

Reply:
Thanks, Schalk! I just checked that localStorage was supported in IE11 and figured that if it failed, it would fail silently. So I assumed that our current check whether it is supported was no longer necessary.

I'll probably just add try/catch here, and then consider converting to use a more generic storage wrapper class if needed later.